src/DataStructures/ComplexDataVector.hpp

```diff
@@ -0,0 +1,179 @@
+                                               MultTrait);
+BLAZE_TRAIT_SPECIALIZE_COMPATIBLE_BINARY_TRAIT(ComplexDataVector, double,
+                                               SubTrait);
+
+template <typename Operator>
+struct BinaryMapTrait<DataVector, ComplexDataVector, Operator> {
+  using Type = ComplexDataVector;
+};
+template <typename Operator>
+struct BinaryMapTrait<ComplexDataVector, DataVector, Operator> {
+  using Type = ComplexDataVector;
+};
+}  // namespace blaze
+
+MAKE_STD_ARRAY_VECTOR_BINOPS(ComplexDataVector)
+
+/// \cond HIDDEN_SYMBOLS
+DEFINE_STD_ARRAY_BINOP(ComplexDataVector, ComplexDataVector,
+                       DataVector, operator+, std::plus<>())
+DEFINE_STD_ARRAY_BINOP(ComplexDataVector, DataVector,
+                       ComplexDataVector, operator+, std::plus<>())
+DEFINE_STD_ARRAY_BINOP(ComplexDataVector, ComplexDataVector, double, operator+,
+                       std::plus<>())
+DEFINE_STD_ARRAY_BINOP(ComplexDataVector, double, ComplexDataVector, operator+,
+                       std::plus<>())
+
+DEFINE_STD_ARRAY_BINOP(ComplexDataVector, ComplexDataVector,
+                       DataVector, operator-, std::minus<>())
+DEFINE_STD_ARRAY_BINOP(ComplexDataVector, DataVector,
+                       ComplexDataVector, operator-, std::minus<>())
+DEFINE_STD_ARRAY_BINOP(ComplexDataVector, ComplexDataVector, double, operator-,
+                       std::minus<>())
+DEFINE_STD_ARRAY_BINOP(ComplexDataVector, double, ComplexDataVector, operator-,
+                       std::minus<>())
+
+DEFINE_STD_ARRAY_INPLACE_BINOP(ComplexDataVector, DataVector, operator+=,
+                               std::plus<>())
+DEFINE_STD_ARRAY_INPLACE_BINOP(ComplexDataVector, DataVector, operator-=,
+                               std::minus<>())
+DEFINE_STD_ARRAY_INPLACE_BINOP(ComplexDataVector, double, operator+=,
+                               std::plus<>())
+DEFINE_STD_ARRAY_INPLACE_BINOP(ComplexDataVector, double, operator-=,
+                               std::minus<>())
+/// \endcond
+
+MAKE_WITH_VALUE_IMPL_DEFINITION_FOR(ComplexDataVector);
+
+namespace ConstantExpressions_detail {
+template <>
+struct pow<ComplexDataVector, 0, std::nullptr_t> {
+  SPECTRE_ALWAYS_INLINE static constexpr std::complex<double> apply(
+      const ComplexDataVector& /*t*/) noexcept {
+    return std::complex<double>{1.0};
+  }
+};
+template <>
+struct pow<std::reference_wrapper<ComplexDataVector>, 0, std::nullptr_t> {
+  SPECTRE_ALWAYS_INLINE static constexpr std::complex<double> apply(
+      const std::reference_wrapper<ComplexDataVector>& /*t*/) noexcept {
+    return std::complex<double>{1.0};
+  }
+};
+template <>
+struct pow<std::reference_wrapper<const ComplexDataVector>, 0, std::nullptr_t> {
+  SPECTRE_ALWAYS_INLINE static constexpr std::complex<double> apply(
+      const std::reference_wrapper<const ComplexDataVector>& /*t*/) noexcept {
+    return std::complex<double>{1.0};
+  }
+};
+
+template <int N>
```

Comment:
is this tested? It looks like the Requires and static_assert are mutually exclusive 😕

Reply:
Hmmm, I agree this is fishy. I've just duplicated this directly from DataVector under the assumption that there was a good (likely blaze-related) reason for these to be here. Now that you point it out, I think I'd prefer to eliminate these and just trust the Blaze pow, unless there's a compelling reason to keep them. @nilsdeppe can you shed some light on what these power functions are used for?

Reply:
Yea, these should be tested. No, you really don't want to defer to a runtime pow, it's horribly slow. I can't remember the exact results but when I compared the computations a year and a half ago at least for small N (less than 10ish or something like that) pow was a factor of several slower than the multiplications. SpEC goes so far as to implement a pow that's runtime but faster if the exponent is an int. We should generally use this version of pow whenever we know the exponent at compile time.

Reply:
Ah, thanks, that helps. Although, I'm still a bit confused. Given the point made by @kidder, and that the self-contradictory implementation was present before I started working on the DataVectors, it seems any compile-time `pow`s currently used in the code base must simply never use negative powers?

I'll add tests, and I think these utilities can be tidied up a bit in the process.

Reply:
So the issue with the implementation is that it allocates memory that doesn't need to be allocated if dividing by a positive exponent. Does that make sense?

Reply:
Ah, got it. I was concerned that it might be causing extra allocations. Great, thanks! I'll see what I can do.